### PR TITLE
mock visit occurrence

### PR DESCRIPTION
**Pull request by ilovemane**

### CODE REVIEW

File: R/checkInputs.R [hunk 1]
```diff
@@ -246,3 +246,31 @@ correctCdmFormat <- function(table, tableName) {
...
 }
+
+#' get occurrence information from cdm table
+#'
+#' @noRd
+#'
+getOccurrence <- function(cdm, tableName, start, end) {
+
+  dataTable <- cdm[[tableName]] |>
+    dplyr::select("person_id", start, end) |>
```
Comment (catalamarti) [inline R/checkInputs.R:257]: start -> dplyr::all_of(start)
end -> dplyr::all_of(end)

File: R/checkInputs.R [hunk 2]
```diff
@@ -246,3 +246,31 @@ correctCdmFormat <- function(table, tableName) {
...
+    dplyr::group_by() |>
+    dplyr::rename(visit_start_date = start,
```
Comment (catalamarti) [inline R/checkInputs.R:259]: you can do the rename at the select stage

File: R/mockVisitOccurrence.R [hunk 3]
```diff
@@ -1,101 +1,96 @@
...
+  }
 
-  concept_id <-
-    cdm$concept |>
-    dplyr::filter(.data$domain_id == "Visit") |>
-    dplyr::select("concept_id") |>
-    dplyr::pull() |>
-    unique()
+  if (length(tableName) == 0) {
+    cli::cli_abort("Your cdm object don't contain clinical tables with visit_occurrence_id.")
```
Comment (catalamarti) [inline R/mockVisitOccurrence.R:34]: no error, just return the cdm the way it is and throw a warning

**Review by catalamarti — APPROVED** [review]

**Review by catalamarti — CHANGES_REQUESTED** [review]
Thanks @ilovemane can you fix the comments I added please?

File: R/checkInputs.R (hunk 1 above)
Comment (ilovemane) [reply R/checkInputs.R:257]: I forgot to delete this

File: R/checkInputs.R (hunk 2 above)
Comment (ilovemane) [reply R/checkInputs.R:259]: this function is not needed

File: R/mockVisitOccurrence.R (hunk 3 above)
Comment (ilovemane) [reply R/mockVisitOccurrence.R:34]: done

**Pull request merged**